gradle-plugin/src/main/kotlin/io/reflekt/plugin/ReflektSubPlugin.kt

```diff
@@ -50,18 +82,21 @@ class ReflektSubPlugin :  KotlinCompilerPluginSupportPlugin {
         version = VERSION
     )
 
-    private fun getFilesToIntrospect(jarFiles: Set<File>): List<File> {
+    private fun getReflektMetaFile(jarFile: File) =
+        extractAllFiles(jarFile).find { it.name == reflektMetaFile } ?: error("Jar file ${jarFile.absolutePath} does not have $reflektMetaFile file!")
+
+    private fun getReflektMetaFiles(jarFiles: Set<File>): List<File> {
         val files: MutableList<File> = ArrayList()
         jarFiles.forEach { jar ->
             getSourceJar(jar)?.let {
-                files.addAll(extractAllFiles(it))
+                files.add(getReflektMetaFile(it))
             }
         }
         return files
     }
 
     private fun getSourceJar(jarFile: File) : File? {
-        val sourceName = "${jarFile.name.substringBeforeLast('.', "")}-sources.jar"
+        val sourceName = "${jarFile.name.substringBeforeLast('.', "")}.jar"
         jarFile.parentFile.parentFile.listFiles()?.filter{ it.isDirectory }?.forEach { folder ->
```

Comment:
`jarFile.parentFile.parentFile` looks really strange.....

Reply:
Why? Probably we should refactor our gradle plugin a bit, since it is old code, I did not change it except the name of the jar

Reply:
But in another PR

Reply:
are we sure each jar has the exact same folder structure so we can call `jarFile.parentFile.parentFile`?

Reply:
I added a TODO about it